### ISSUE TITLE
Merge `ape-template` into core

### ISSUE THREAD

**Issue by fubuloubu** [ApeWorX/ape#1556]
### Overview

https://github.com/cookiecutter/cookiecutter is now actively supported. We we originally prototyped the `cookiecutter` integration when that package was not being actively supported by it's original author, so we weren't sure if the package was going to be well-supported in the future. However, now it is supported.

`ape-template`'s feature are better off as core functionality, and we should investigate integrating them as a core plugin so that we can leverage it more for other functionality (such as enhancing `ape init`'s default template with additional questions for project setup h/t @pandadefi)

Also found `copier`, which is an alternative project we could migrate to if needed:
https://github.com/ApeWorX/ape-template/issues/9

### Specification

Not final, but some ideas of where we can take it once merged:

```sh
$ ape init
# Does the same thing it does today, only interactively asking more questions about setup
$ ape init --name "Project Name"  # etc...
# Overloads the template's interactive questions with pre-defined answers
$ ape init gh:ApeAcademy/ERC721
# Use existing `ape-template` functionality to clone a target template
```

### Dependencies
Requires retiring `ape-template` and updating Ape Academy content to match

**Comment by antazoey** [comment]
@fubuloubu Are you still wanting this? I added the 0.8 tag.

**Comment by fubuloubu** [comment]
Yes, I think it's important to have this functionality in Core, and my concerns with cookiecutter are no longer an issue as they are more actively developed now

**Comment by antazoey** [comment]
Cool, I am thinking when that is done, it will also close https://github.com/ApeWorX/ape/issues/1816, provided the docs are good.

**Comment by fubuloubu** [comment]
> Cool, I am thinking when that is done, it will also close https://github.com/ApeWorX/ape/issues/1816, provided the docs are good.

Yes exactly, should be better able to resolve that issue with some better docs and a few good templates to choose from

**Comment by antazoey** [comment]
looking at ape-template, it doesnt really do anything. all it does is re-expose the cookiecutter cli.
I am not 100% sure what needs to be done here.
